docs/docs-beta/docs/code_examples/guides/automation/asset-sensor-custom-eval.py

```diff
@@ -0,0 +1,51 @@
+from dagster import (
+    AssetExecutionContext,
+    AssetKey,
+    AssetMaterialization,
+    Definitions,
+    MaterializeResult,
+    RunRequest,
+    SensorEvaluationContext,
+    SkipReason,
+    asset,
+    asset_sensor,
+    define_asset_job,
+)
+
+
+@asset
+def daily_sales_data(context: AssetExecutionContext):
+    context.log.info("Asset to watch, perhaps some function sets metadata here")
+    yield MaterializeResult(metadata={"specific_property": "value"})
+
+
+@asset
+def weekly_report(context: AssetExecutionContext):
+    context.log.info("Running weekly report")
+
+
+my_job = define_asset_job("my_job", [weekly_report])
+
+
+@asset_sensor(asset_key=AssetKey("daily_sales_data"), job=my_job)
```

Comment:
What are we establishing as best practice for asset key selection?

`AssetKey("daily_sales_data")` is likely best, but I find that using strings can result in bugs as we don't have auto completion. Whereas `daily_sales_data.key` can result in code location load error if misspelled.

Reply:
I would have preferred `"daily_sales_data"` to be honest, but asset_sensor doesnt take a `Coercible`. Esepcially for asset sensors, using `daily_sales_data.key` would not work since you might not be importing the asset itself.